force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,34 @@
+trigger AccountTrigger on Account (before insert, after insert) {
+    
+    if (Trigger.isBefore) {
+        
+        for (Account acc : Trigger.new) {
+            // Q1
+            if (acc.Type == null) {
+                acc.Type = 'Prospect';
+            }
+
+            // Q2
+            acc.BillingStreet = acc.ShippingStreet ?? null;
+            acc.BillingCity = acc.ShippingCity ?? null;
+            acc.BillingState = acc.ShippingState ?? null;
+            acc.BillingPostalCode = acc.ShippingPostalCode ?? null;
+            acc.BillingCountry = acc.ShippingCountry ?? null;
```

Comment:
I appreciate the use of the null coalescing operator, but if you are checking it is not null and then assigning it to null anyways, probably not necessary. Warren and I disagree on this, but I would check that if Billing was null, then assign shipping. If they have provided a billing, dont overwrite it, otherwise assign the shipping. Also addresses are complete units, so you would not want to only set the state if not null, you want to take the whole address when not supplied. None of this is relevant for answering this homework really, but something to think about in the real world.

Reply:
Yes... I was so focused on the HW instructions and took it very literally, but toally agree with what you're saying.